rm/src/lib.rs

```diff
@@ -17,82 +19,157 @@ pub const HELP: Help = Help {
     usage: r#"
 Usage
 
-    rad rm <urn> [<option>...]
+    rad rm <urn | profile-id> [<option>...]
 
 Options
 
-    -i        Prompt before removal
-    --help    Print help
+    --no-confirm        Do not ask for confirmation before removal
+                        (default: false)
+    --no-passphrase     If profile id given, bypass passphrase prompt and
+                        do not read environment variable `RAD_PASSPHRASE`
+                        nor standard input stream (default: false)
+    --stdin             Read passphrase from stdin (default: false)
+    --help              Print help
 "#,
 };
 
+enum Object {
+    Project(Urn),
+    Profile(ProfileId),
+    Unknown(String),
+}
+
+impl From<&str> for Object {
+    fn from(value: &str) -> Self {
+        if let Ok(urn) = Urn::from_str(value) {
+            Object::Project(urn)
+        } else if let Ok(id) = ProfileId::from_str(value) {
+            Object::Profile(id)
+        } else {
+            Object::Unknown(value.to_owned())
+        }
+    }
+}
+
 pub struct Options {
-    urn: Urn,
-    prompt: bool,
+    object: Object,
+    confirm: bool,
+    require_secret: bool,
+    stdin: bool,
 }
 
 impl Args for Options {
     fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
         use lexopt::prelude::*;
 
         let mut parser = lexopt::Parser::from_args(args);
-        let mut urn: Option<Urn> = None;
-        let mut prompt = false;
+        let mut object: Option<Object> = None;
+        let mut confirm = true;
+        let mut require_secret = true;
+        let mut stdin = false;
 
         while let Some(arg) = parser.next()? {
             match arg {
-                Short('i') => {
-                    prompt = true;
+                Long("no-confirm") => {
+                    confirm = false;
+                }
+                Long("no-password") => {
+                    require_secret = false;
```

Comment:
I find it easier to use the same names as the options for these vars, that way it's clear what they represent, so just `passphrase = false`